README.md

```diff
@@ -43,7 +45,30 @@ Generate a project from the `pybamm-cookiecutter` template.
 **Note:** This requires an internet connection. You could manually clone the git repository and run the copy command, or just execute the copy command with the URL to the git repository.
 
 ```bash
-copier copy https://github.com/pybamm-team/pybamm-cookiecutter.git name_of_your_project/ --trust
+copier copy https://github.com/pybamm-team/pybamm-cookiecutter.git . --trust
+# this will generate the project in the current working directory
+copier copy https://github.com/pybamm-team/pybamm-cookiecutter.git path_to_copy_to/ --trust
+# this will generate the project in the specified path
+```
+#### Using pipx
+
+Clone the `pybamm-cookiecutter` github repository.
+```bash
+git clone https://github.com/pybamm-team/pybamm-cookiecutter.git
+```
+Navigate into the `pybamm-cookiecutter` and use `pipx` to access the `CLI`.
+```bash
+cd pybamm-cookiecutter
+pipx install .
+```
+Once `pipx` installs the package locally, you can execute it to generate a project.
+```bash
+pybamm_cookiectter
+# this will prompt the user for project configuration and generate
+# the project in the current working directory.
+# or
+pybamm_cookiecutter --path /path_to_copy_to
+# this will install the project in the specified path.
```

Comment:
I think we should recommend generating the project in a separate directory from the first go since running it in the current working directory (where the `pybamm-cookiecutter` project exists) will raise many conflicts. Either way, I think we should remove the git clone step and assume that we are writing the documentation in a way that suggests that `pybamm-cookiecutter` is already on PyPI, given that we are publishing it pretty soon.

Reply:
No, that will not be the case with the CLI when installed through `pipx`, it is installed system-wide and hence there won't necessarily be a case where a user generates the project inside the `pybamm-cookiecutter` project unless they try to install it from the source and execute the command while being inside the project directory unknowingly. 
In most cases, a user would perform - 
```bash 
pipx install pybamm-cookiecutter
# Assuming that I am in my home directory `/home` and I give the project name pybamm-test, it will look something like this - 
pybamm-cookiecutter # user prompt initialises and I give the project name as mentioned above
```
`pybamm-test` will be generated like `home/pybamm-test/src/pybamm_test/...`, but obviously it is up to the user to provide the `--path` argument as well.
I think your main concern is executing this command in an unknown directory and all of the project files populating themselves in that directory creating a mess which is why I added a directory restructure within this PR which would automatically create a directory with the `project_name` and populate all the files inside it, so if it is a mistake a user can easily just delete the parent directory, in this case `pybamm-test`. 

In case they opt to provide the path argument and want to have a generated project inside the `home/test/` path, the generated project path would look something like this -
```bash 
pybamm-cookiecutter --path /home/test 
```
The generated project path would be `home/test/pybamm-test/src/pybamm_test`.

I think the documentation part I agree with writing it in the way which resembles the behaviour of a `pipx` `pybamm-cookiecutter` package.  I would love to hear what @brosaplanella thinks as well as I mostly added the documentation for immediate setup purposes for his projects.

Reply:
Ah, apologies if I was too terse, let me clarify:

> No, that will not be the case with the CLI when installed through `pipx`, it is installed system-wide and hence there won't necessarily be a case where a user generates the project inside the `pybamm-cookiecutter` project

Indeed, the usage instructions were instructing the user to run it with `pipx` in the git-clone. You have now updated this in da1e56666f4e452c73e58284af28b2e36b2520c7, which is fine with me.

> I think your main concern is executing this command in an unknown directory and all of the project files populating themselves in that directory creating a mess

Yes, I was under the impression that a user would execute the command either inside the directory where they want their project, or that they would let the command create the directory. But, I don't expect a user to run it inside a non-empty directory in the first place, so there shouldn't be any conflicts with your new approach.

> which is why I added a directory restructure within this PR which would automatically create a directory with the project_name and populate all the files inside it

Sounds good!

My only point of contention was that developers should not use `pipx` to install the project from source (whether in editable mode or not); a standard `pip install` would be better – this is already covered by `nox`, so there's nothing more to add.

The other point was that we are quite close to the PyPI release, and therefore, we should remove notes about the project not being on PyPI yet in the paragraph above this one.

Reply:
I missed the directory restructure commit in my last `copier` migration PR which would have avoided all the confusion haha and should've made it clearer, my mistake apologies! 😅

Reply:
No worries, thanks for fixing that 🙏🏻